test/src/ui/pages/registration/registration_notifier_test.dart

```diff
@@ -0,0 +1,108 @@
+  MockExceptionMapper mockExceptionMapper;
+
+  setUp(() {
+    mockAuthenticationRepository = MockAuthenticationRepository();
+    mockExceptionMapper = MockExceptionMapper();
+    registrationNotifier = RegistrationNotifier(mockAuthenticationRepository, mockExceptionMapper);
+
+    registrationNotifier.setPassword("mock");
+    registrationNotifier.setEmail("mock");
+    registrationNotifier.setName("mock");
+  });
+
+  group('notifier tests', () {
+    test('should emit loaded registration state when registering new user', () {
+      when(mockAuthenticationRepository.createUser(any, any)).thenAnswer((_) async => Future.value());
+      when(mockAuthenticationRepository.updateUser(any)).thenAnswer((_) async => Future.value());
+
+      registrationNotifier.verifyCreateUser(
+        registrationNotifier.createUser,
+        [
+          () {
+            expect(registrationNotifier.state.isLoading, true);
+          },
+          () {
+            verify(mockAuthenticationRepository.createUser(any, any));
+            verify(mockAuthenticationRepository.updateUser(any));
+            expect(registrationNotifier.state.isLoading, false);
+          }
+        ],
+      );
+    });
+  });
+
+  group('should return correct error when registering the user', () {
+    final inputsToMatchers = {
+      FirebaseAuthException(code: _Constants.tooManyRequests, message: ""): TooManyRequests(""),
+      FirebaseAuthException(code: _Constants.invalidEmail, message: ""): InvalidEmailException(""),
+      FirebaseAuthException(code: _Constants.networkError, message: ""): NetworkException(""),
+      FirebaseAuthException(code: _Constants.userNotFound, message: ""): UserNotFoundException(""),
+      FirebaseAuthException(code: _Constants.operationNotAllowed, message: ""): OperationNotAllowedException(""),
+      FirebaseAuthException(code: _Constants.emailAlreadyInUse, message: ""): EmailAlreadyInUseException(""),
+      FirebaseAuthException(code: _Constants.userDisabled, message: ""): UserDisabledException(""),
+      FirebaseAuthException(code: _Constants.wrongPassword, message: ""): WrongPasswordException(""),
+      FirebaseAuthException(code: _Constants.weakPassword, message: ""): WeakPasswordException(""),
+    };
+
+    inputsToMatchers.forEach((input, matcher) {
+      test("$input -> $matcher", () {
+        when(mockAuthenticationRepository.createUser(any, any)).thenAnswer((_) async => Future.error(input));
+        when(mockExceptionMapper.toDomainException(any)).thenAnswer((_) => matcher);
+
+        registrationNotifier.verifyCreateUser(
+          registrationNotifier.createUser,
+          [
+            () {
+              expect(registrationNotifier.state.isLoading, true);
+            },
+            () {
+              expect(registrationNotifier.state.exception.runtimeType, matcher.runtimeType);
+            }
+          ],
+        );
+      });
+    });
+  });
+}
+
+extension on ChangeNotifier {
+  void verifyCreateUser(
```

Comment:
jak nie będziesz niepotrzebnie przesyłał parametrów do metod create i signIn to nie będą potrzebne te extensiony :)